### ISSUE TITLE
Public collections should only show one version

### ISSUE THREAD

**Issue by nguyenLisa** [nguyenLisa/abm#40]
A public collection should only display the version that was made public, and that version only.
On the view that details a collection:
- The version drop-down should not be shown for the pages of a public collection.
- The ID of the version and creation date should be shown in the collection information.
- When making the collection public, the confirmation pop-up should provide the version drop-down with the default entry set to the version that is currently displayed on the screen, so that the user can choose which version to make public.

**Comment by ankur-1989** [comment]
@nguyenLisa 
We are showing "Derive new version" in disabled form for public collection when nobody is logged in, but if I am logged in into the application it is enabled. Do we need to keep it disabled always for public collection?

As of now, It makes the complete collection public, we will have to make changes to make the collection public according to version.
I'll assign it to myself and will make the changes accordingly.

**Comment by ankur-1989** [comment]
In this change, we have made the following design changes in making public functionality - 
1. Now onwards, Collection will always remain private. 
2. We will make collection public version-wise. 
3. User will not be able to edit the public version, even if the user is the owner of the collection.
4. User can see build results for public version when the user is logged in into the application.
5. On Public Collections page, it will show a list of those collections which has public versions. 
6. when you open a particular collection, then it will show only public versions instead of all the versions.

@nguyenLisa 
 Are all the points correct?  Do we need to make any change anywhere?

**Comment by nguyenLisa** [comment]
Thanks for taking over this.
1. Ok
2. Ok
3. Ok, but there may be multiple public versions.
4. Ok
5. Ok
6. Ok, except when you are the collection owner and you are logged in. Then, you can see all versions of the collection.

**Comment by ankur-1989** [comment]
Yes, Point number 3 and 6 have been taken care as you mentioned.

**Comment by ankur-1989** [comment]
Changes related to this issue have been implemented and pushed to develop branch of abm-frontend repository. Following technical changes have been done as part of it.

1. A new column in version table with the name privateStatus has been added.
2. privateStatus check has been shifted to version from collection in Buildcontroller.java and HermesController.java files for download functionality.
3. Join with versions has been applied in JpaCollectionDao.java file in order to find those collections which have public versions.
4. privateStatus column has been added in JpaVersionDao.java , JpaVersion.java and VersioDTO.java files at respective locations.
5. Respective changes have been made in editCollection.component.ts and viewCollection.component.ts files in order to show the collections with public versions.
